runtime/compiler/control/CompilationThread.cpp

```diff
@@ -7427,10 +7486,13 @@ TR::CompilationInfoPerThreadBase::preCompilationTasks(J9VMThread * vmThread,
 #endif /* defined(J9VM_OPT_JITSERVER) */
          _vm = TR_J9VMBase::get(_jitConfig, vmThread, TR_J9VMBase::AOT_VM);
 #if defined(J9VM_OPT_JITSERVER)
-      if ((_compInfo.getPersistentInfo()->getRemoteCompilationMode() == JITServer::CLIENT) &&
-         !shouldPerformLocalComp(entry))
+      if (_compInfo.getPersistentInfo()->getRemoteCompilationMode() == JITServer::CLIENT)
          {
-         entry->setRemoteCompReq();
+         bool forcedLocal;
+         if (!shouldPerformLocalComp(entry, forcedLocal))
+            entry->setRemoteCompReq();
+         else
+            downgradeLocalCompilationIfLowPhysicalMemory(entry, forcedLocal);
```

Comment:
I am thinking that it may be more explicit if we perform a test for `forcedLocal` outside `donwngradeLocalCompilationIfLowPhysicalMemory`. This way we avoid entering that method for nothing and the name of the method better reflects the outcome (as opposed to sometimes downgrade local compilations)